### DISCUSSION TITLE
Memory deallocation when using emscripten_async_wget_data

### DISCUSSION THREAD

**Discussion by sasmaster** [emscripten-core/emscripten#18277] [Q&A]
I have a question related to the memory allocation when using `emscripten_async_wget_data` .Do I have to free the buffer returned in the callback of that function?

**Comment by kripken** [comment]
[The docs](https://emscripten.org/docs/api_reference/emscripten.h.html#c.emscripten_async_wget_data) say this:

> (void)* : A pointer to a buffer with the data. Note that, as with the worker API, the data buffer only lives during the callback; it must be used or copied during that time.

So you do not need to free it.
